pkg/option/config_test.go

```diff
@@ -700,6 +700,151 @@ func Test_populateNodePortRange(t *testing.T) {
 	}
 }
 
+func Test_populateIdentityAllocation(t *testing.T) {
+	type want struct {
+		wantIdentityAllocationMode   string
+		wantKVStore                  string
+		wantDisableCiliumEndpointCRD bool
+		wantErr                      bool
+	}
+	tests := []struct {
+		name       string
+		want       want
+		preTestRun func()
+	}{
+		{
+			name: "no identity allocation mode, no kvstore provided, should default to CRD",
+			want: want{
+				wantIdentityAllocationMode:   IdentityAllocationModeCRD,
+				wantKVStore:                  "",
+				wantDisableCiliumEndpointCRD: false,
+				wantErr:                      false,
+			},
+			preTestRun: func() {
+				viper.Reset()
+				viper.Set(IdentityAllocationMode, "")
+				viper.Set(KVStore, "")
+			},
+		},
+		{
+			name: "no identity allocation mode specified, kvstore type provided, should set kvstore mode",
+			want: want{
+				wantIdentityAllocationMode:   IdentityAllocationModeKVstore,
+				wantKVStore:                  "etcd",
+				wantDisableCiliumEndpointCRD: false,
+				wantErr:                      false,
+			},
+			preTestRun: func() {
+				viper.Reset()
+				viper.Set(IdentityAllocationMode, "")
+				viper.Set(KVStore, "etcd")
+			},
+		},
+		{
+			name: "crd mode, kvstore type provided, should unset kvstore type",
+			want: want{
+				wantIdentityAllocationMode:   IdentityAllocationModeCRD,
+				wantKVStore:                  "",
+				wantDisableCiliumEndpointCRD: false,
+				wantErr:                      false,
+			},
+			preTestRun: func() {
+				viper.Reset()
+				viper.Set(IdentityAllocationMode, IdentityAllocationModeCRD)
+				viper.Set(KVStore, "etcd")
+			},
+		},
+		{
+			name: "crd mode provided, endpoint crd enabled, should enable endpoint crd",
+			want: want{
+				wantIdentityAllocationMode:   IdentityAllocationModeCRD,
+				wantKVStore:                  "",
+				wantDisableCiliumEndpointCRD: false,
+				wantErr:                      false,
+			},
+			preTestRun: func() {
+				viper.Reset()
+				viper.Set(IdentityAllocationMode, IdentityAllocationModeCRD)
+				viper.Set(DisableCiliumEndpointCRDName, true)
+			},
+		},
+		{
+			name: "kvstore mode, no kvstore type provided, should set crd mode",
+			want: want{
+				wantIdentityAllocationMode:   IdentityAllocationModeCRD,
+				wantKVStore:                  "",
+				wantDisableCiliumEndpointCRD: false,
+				wantErr:                      false,
+			},
+			preTestRun: func() {
+				viper.Reset()
+				viper.Set(IdentityAllocationMode, IdentityAllocationModeKVstore)
+				viper.Set(KVStore, "")
+			},
```

Comment:
Hmm, I kinda wonder if someone explicitly passes kvstore mode but fails to properly configure it whether it should actually error out..?